backend/tracim_backend/command/cleanup.py

```diff
@@ -0,0 +1,248 @@
+            "created content in other user workspaces, this may create inconsistent database.",
+            dest="force",
+            default=False,
+            action="store_true",
+        )
+        parser.add_argument(
+            "-l", "--login", nargs="+", help="User logins (email)", dest="logins", required=True
+        )
+        parser.add_argument(
+            "--anonymous-name",
+            help="Anonymous user display name to use",
+            dest="anonymous_name",
+            required=False,
+        )
+        return parser
+
+    def take_app_action(self, parsed_args: argparse.Namespace, app_context: AppEnvironment) -> None:
+        self._session = app_context["request"].dbsession
+        self._app_config = app_context["registry"].settings["CFG"]
+
+        if parsed_args.dry_run_mode:
+            print("(!) Running in dry-run mode, no changes will be applied.")
+        if parsed_args.force:
+            print("/!\\ Running in force mode, database created may be broken /!\\.")
+        elif parsed_args.best_effort:
+            print(
+                "(!) Running in best effort mode, will anonymise account instead of deleting them if needed."
+            )
+
+        deleted_user_ids = set()
+        deleted_workspace_ids = set()
+
+        with unprotected_content_revision(self._session) as session:
+            uapi = UserApi(
+                config=self._app_config,
+                session=session,
+                current_user=None,
+                show_deleted=True,
+                show_deactivated=True,
+            )
+            user_list = []  # type: typing.List[User]
+            for login in parsed_args.logins:
+                try:
+                    user = uapi.get_one_by_email(login)
+                    user_list.append(user)
+                except UserDoesNotExist:
+                    print("ERROR: user with email {} does not exist".format(login))
+                    exit(1)
+            print("~~~~~~~~~~")
+            for user in user_list:
+                cleanup_lib = CleanupLib(
+                    session, self._app_config, dry_run_mode=parsed_args.dry_run_mode
+                )
+                deleted_user_ids_result = self._delete_user_database_info(
+                    user,
+                    force=parsed_args.force,
+                    best_effort=parsed_args.best_effort,
+                    cleanup_lib=cleanup_lib,
+                    anonymised_user_display_name=parsed_args.anonymous_name,
+                )
+                deleted_user_ids.add(deleted_user_ids_result.user_id)
+                deleted_workspace_ids.update(deleted_workspace_ids)
+                print("~~~~~~~~~~")
+
+            # INFO - G.M - 2019-12-13 - cleanup agenda at end of process
+            if deleted_workspace_ids:
+                deleted_workspace_ids_str = [
+                    '"{}"'.format(workspace_id) for workspace_id in deleted_workspace_ids
+                ]
```

Comment:
These lines are only for str conversion of `deleted_workspace_ids` ?

If yes, you can do:

``` python
deleted_workspace_ids_str = list(map(str, deleted_workspace_ids))
```

Reply:
`list(` is not required as you only browse it

Reply:
Not sure, we convert 1(int) to "1" string (including quote in the string.